## Assignment # 10 - Point Estimate and Interval Estimate (Confidence Interval)

A random survey of enrollment at 35 community colleges across the United States yielded the following figures:

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;

5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;

17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;

1,263; 7,285; 28,165; 5,080; 11,622

Perform point estimate and interval estimate with 95% confidence level using t-distribution.

Since we don't know the population variance, we use t-distribiution instead of normal distribution.

In [27]:
# Import Python packages
import math
from scipy.stats import t

In [2]:
# make each line of numbers a string object and then concatenate them together 
# The end result is one single string containing 35 numbers separated by ";"

data_1 = "6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;" 
data_2 = "5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; "
data_3 = "17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; "
data_4 = "1,263; 7,285; 28,165; 5,080; 11,622"
data = data_1 + data_2 + data_3 + data_4
print(data)

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; 17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; 1,263; 7,285; 28,165; 5,080; 11,622


In [3]:
# Convert the single string to a list of strings using split() function
# Make sure to specify a delimter or separator

x = data.split(";")
print(x)

['6,414', ' 1,550', ' 2,109', ' 9,350', ' 21,828', ' 4,300', ' 5,944', ' 5,722', ' 2,825', ' 2,044', '5,481', ' 5,200', ' 5,853', ' 2,750', ' 10,012', ' 6,357', ' 27,000', ' 9,414', ' 7,681', ' 3,200', ' 17,500', ' 9,200', ' 7,380', ' 18,314', ' 6,557', ' 13,713', ' 17,768', ' 7,493', ' 2,771', ' 2,861', ' 1,263', ' 7,285', ' 28,165', ' 5,080', ' 11,622']


In [4]:
e = [int(i.replace(",", "")) for i in x]
print(e)

[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


In [8]:
# Create a list of integers from the list of strings using List Comprehension or a function
# Make sure to remove the "," first and then convert the strings to integers

e = []
for i in x:
    y = i.replace(",", "")
    y = int(y)
    e.append(y)
print(e)

[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


### Step 1 - Calculate and Display the Sample Size and Sample Mean

In [16]:
# Calculate and display the sample size (sample size = 35)
size = 0 
for index in range(len(e)):
  size = size + 1
print("Sample size =" , size)

Sample size = 35


In [17]:
# Calculate and display the sample mean (sample mean = 8629.0)
sum = 0
for index in range(len(e)):
  sum = sum + e[index]

sample_mean = sum / size
print("Sample mean =", round(sample_mean))

Sample mean = 8629


The point estimate of the mean enrollment of US community colleges is 8629.

### Step 2 - Calculate and Display the Sample Standard Deviation & Sample Standard Error

Sample Standard Deviation $S=\sqrt{\dfrac{1}{n-1}\sum\limits_{i=1}^n (X_i-\bar{X})^2}$

Sample Standard Error = $\dfrac{S}{\sqrt{n}}$

Note: The default Delta Degree of Freedom (DDOF) for Numpy's std function is 0 which is applicable to populate data. For sample data, we need to specify ddof=1.

For the enrollment data, we round up the statistics to be the full integers (no decimal points).

In [24]:
# Calculate and display the sample standard deviation using Numpy's std function. (sample standard deviation = 6944.0)
sum = 0
for index in range(len(e)):
  temp = (e[index] - sample_mean)**2
  sum = sum + temp
sample_sd = math.sqrt(  (  (1/(size-1)) * sum   )   )
print("Sample standard deviation =" , round(sample_sd))

Sample standard deviation = 6944


In [26]:
# Calculate and display the sample standard error (sample standard error = 1174)
sample_se = sample_sd / (math.sqrt(size))
print("Sample standard error =" , round(sample_se))

Sample standard error = 1174


### Step 3 - Calculate t Critical Value using t-Distribution

$\alpha$ = 1 - Confidence Level = 1 - 95% = 0.05

$\dfrac{\alpha}{2}$ = 0.025

n (sample size) = 35

df (degree of freedom) = n - 1 = 35 - 1 = 34

We will use Python scipy.stats t-distribution's PPF (Percentage Point Function) to calculate t critical value $t_{0.025,34}$.

In [46]:
# Calculate and display the t critical value using scipy.stats.t package ppf function (t critical value = 2.03)
t_value = abs(t.ppf(.025,34))
print("t critical value =" , round(t_value,2))

t critical value = 2.03


### Step 4 - Calculate the Margin of Error

Margin of Error = t-Statistics * Sample Standard Error = $t_{\alpha/2,n-1}\left(\dfrac{s}{\sqrt{n}}\right)$

In [48]:
# Calculate and display the margin of error (margin of error = 2383.0)
margin_error = round(t_value,2) * sample_se
print("Margin of error = " , round(margin_error))

Margin of error =  2383.0


### Step 5 - Calculate Lower and Upper Limit of the Confidence Interval

Lower Limit = Sample Mean - Margin of Error

Upper Limit = Sample Mean + Margin of Error

In [49]:
# Calculate and display the lower limit (lower limit = 6246.0)
lower_limit = sample_mean - margin_error
print("Lower limit = ", round(lower_limit))

Lower limit =  6246.0


In [50]:
# Calculate and display the upper limit (upper limit = 11012.0)
upper_limit = sample_mean + margin_error
print("Upper limit = ", round(upper_limit))

Upper limit =  11011.0


### Step 6 - Now We have the 95% Confidence Interval

Confidence Interval ($\sigma$ unknown) = $\bar{x}\space\pm\space t_{\alpha/2}\left(\dfrac{S}{\sqrt{n}}\right)$ = Sample_Mean $\pm$ Margin of Error

In [52]:
# Print the 95% confidence interveral in the form (lower limit, upper limit) (the 95% confidence interval = 6246.0 , 11012.0)
print("The 95% confidence interveral = (",round(lower_limit),",",round(upper_limit),")")

The 95% confidence interveral = ( 6246.0 , 11011.0 )


Complete :)